src/hub.js

```diff
@@ -297,16 +299,25 @@ const onReady = async () => {
     });
 
     const offset = { x: 0, y: 0, z: -1.5 };
-    const spawnMediaInfrontOfPlayer = src => {
-      const entity = addMedia(src, true);
+    const spawnMediaInfrontOfPlayer = (src, contentOrigin) => {
+      const entity = addMedia(src, contentOrigin, true);
+
       entity.setAttribute("offset-relative-to", {
         target: "#player-camera",
         offset
       });
     };
 
     scene.addEventListener("add_media", e => {
-      spawnMediaInfrontOfPlayer(e.detail);
+      const origin = e.detail instanceof File ? ObjectContentOrigins.FILE : ObjectContentOrigins.URL;
```

Comment:
Would call this `contentOrigin` to avoid confusion with `origin` from the media response.